projects/explorer/src/app/views/blocks/blocks.component.html

```diff
@@ -1,32 +1,51 @@
-<h2>Blocks</h2>
-<div>
-  <mat-card>
-    <ng-container *ngIf="blocks === undefined || null; then empty; else exist"></ng-container>
-    <ng-template #empty></ng-template>
-    <ng-template #exist>
-      <mat-list>
-        <mat-list-item>
-          <span class="break-all truncate pr-4">Height</span>
+<div class="flex flex-row justify-between content-start items-start">
+  <h2>Latest Blocks</h2>
+  <span class="flex-auto"></span>
+  <mat-checkbox class="auto" color="primary" (change)="onCheckBoxAutoChange($event.checked)">
+    Auto
+  </mat-checkbox>
+</div>
+
+<mat-card>
+  <ng-container *ngIf="blocks === undefined || null; then empty; else exist"></ng-container>
+  <ng-template #empty></ng-template>
+  <ng-template #exist>
+    <mat-list>
+      <mat-list-item>
+        <span class="break-all truncate pr-4">Height</span>
+        <span class="flex-auto"></span>
+        <span class="w-1/2 break-all truncate pl-4">Proposer</span>
+        <span class="flex-auto"></span>
+        <span class="w-1/8 break-all truncate">Num. Tx</span>
+        <span class="flex-auto"></span>
+        <span class="w-1/4 break-all truncate">Time</span>
+        <mat-divider></mat-divider>
+      </mat-list-item>
+    </mat-list>
+    <mat-nav-list>
+      <ng-container *ngFor="let block of blocks">
+        <mat-list-item routerLink="/blocks/{{ block?.block?.header?.height }}">
+          <span class="w-1/8 pr-4">{{ block?.block?.header?.height }}</span>
+          <span class="flex-auto"></span>
+          <span class="w-1/2 break-all">{{ block?.block?.header?.proposer_address }}</span>
```

Comment:
I guess proposer_address should be converted to properly type. (I guess valAddress string. But I'm not sure...)
It would be better to create a separate new Issue and handle it there.

Reply:
I create the following issue. Let's handle it there.
https://github.com/UnUniFi/web-apps/issues/58